### PR TITLE
feat: social media sharing

### PR DESCRIPTION
**Pull request by andyv09**

### CODE REVIEW

File: packages/dapp/src/messages/en.json [hunk 1]
```diff
@@ -537,6 +537,7 @@
     "selected": "Selected Credentials",
     "share-presentation-error": "Failed to share presentation",
     "share-presentation-success": "Successfully shared presentation",
+    "share-link-description": "Share your credential on social media:",
```
Comment (Vid201) [inline packages/dapp/src/messages/en.json:540]: Remove "on social media"

**Comment by pseudobun** [comment]
Fix prettier warnings.
```
[warn] src/app/[locale]/app/(public)/share-presentation/[id]/credentialPanel.tsx
[warn] src/app/[locale]/app/(public)/share-presentation/[id]/formatedView.tsx
[warn] src/app/[locale]/app/(public)/share-presentation/[id]/page.tsx
[warn] src/components/ShareCredentialModal/index.tsx
[warn] Code style issues found in 4 files. Run Prettier to fix.
```

File: packages/dapp/src/app/[locale]/app/(public)/share-presentation/[id]/page.tsx [hunk 2]
```diff
@@ -85,3 +86,74 @@ export default async function Page({
...
+    view: 'Normal' | 'Json';
+    page: string | undefined;
+  };
+}) {
+  const { presentation, title } = await getPresentation(id);
+
+  if (!presentation) return {};
+
+  // const url = process.env.NEXT_PUBLIC_APP_URL ?? 'https://masca.io';
+  const url = 'http://localhost:3000';
```
Comment (pseudobun) [inline packages/dapp/src/app/[locale]/app/(public)/share-presentation/[id]/page.tsx:105]: Prepare this for deployment.

File: packages/dapp/src/app/api/og/route.tsx [hunk 3]
```diff
@@ -24,8 +24,222 @@ export async function GET(req: NextRequest) {
...
+      holder,
+      numberOfCredentials,
+      credentialIssuer = 'unknown',
+      credentialType = 'unknown',
+      credentialSubject = 'unknown',
+      credentialIssuanceDate = 'unknown',
     } = values;
 
+    if (type && type === 'share-presentation') {
+      // If 1 VC show VC
```
Comment (pseudobun) [inline packages/dapp/src/app/api/og/route.tsx:37]: Remove this comment.

File: packages/dapp/src/app/api/og/route.tsx [hunk 4]
```diff
@@ -24,8 +24,222 @@ export async function GET(req: NextRequest) {
...
+                data: fontBold,
+                weight: 700,
+                style: 'normal',
+              },
+            ],
+          }
+        );
+      }
+
+      // If more...
```
Comment (pseudobun) [inline packages/dapp/src/app/api/og/route.tsx:154]: This one too.

**Review by pseudobun — CHANGES_REQUESTED** [review]

**Comment by pseudobun** [comment]
Resolve prettier issues:
```
./src/app/[locale]/app/(public)/share-presentation/[id]/page.tsx
1:10  Error: 'ResolvingMetadata' is defined but never used.  unused-imports/no-unused-imports
```

File: packages/dapp/src/app/api/og/route.tsx [hunk 5]
```diff
@@ -24,8 +24,327 @@ export async function GET(req: NextRequest) {
...
+                  <div
+                    tw="tracking-tight flex text-3xl text-white"
+                    style={{
+                      fontFamily: 'Inter',
+                      fontWeight: 'normal',
+                    }}
+                  >
+                    {holder.substring(0, 20)}...
+                    {holder.substring(holder.length, holder.length - 10)}
+                    di
```
Comment (pseudobun) [inline packages/dapp/src/app/api/og/route.tsx:301]: Is this supposed to be here?

**Review by pseudobun — CHANGES_REQUESTED** [review]

**Pull request merged**